proxy/proxyports: move test-only code and use fake datapath iptables manager

tklauser opened the pull request:
Currently `proxyPortsForTest` is defined in `testutils.go` which will also
be compiled for non-test builds (due to the lack of a `_test.go` file
suffix), although it's only used in test code. Move it to
`proxyports_test.go` along with the other test code.

Also replace `MockIPTablesManager` by the same fake/mock implementation
already defined in `pkg/datapath/fake/types.FakeIptablesManager`.

mhofstetter commented on pkg/proxy/proxyports/proxyports.go:
```diff
@@ -119,7 +118,7 @@ type ProxyPorts struct {
 	logger *slog.Logger,
 	config ProxyPortsConfig,
-	datapathUpdater datapath.IptablesManager,
+	datapathUpdater DatapathUpdater,
```
```
time=2025-07-22T10:12:27.143263331Z level=error source=/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/cell/invoke.go:52 msg="Invoke failed" error="missing type:\n\t- proxyports.DatapathUpdater (did you mean to Provide it?)" function="cmd.configureAPIServer (cmd/cells.go:347)"
```

`DatapathUpdater` is not provided. Either we need to provide that type or need to extract an internal function (that uses this type in its signature) that can be used for unit-testing purposes.

tklauser replied:
Sorry, `ENOTENOUGHSLEEP` :facepalm: that's probably not worth the churn then. I'll drop the commit.

tklauser replied:
Turns out we can get rid of `MockIPTablesManager` altogether and use `pkg/datapath/fake/types.FakeIptablesManager`. Updated accordingly.

tklauser committed:
proxy/proxyports: move test-only code and use datapath FakeIptablesManager

Currently proxyPortsForTest is defined in testutils.go which will also
be compiled for non-test builds (due to the lack of a _test.go file
suffix), although it's only used in test code. Move it to
proxyports_test.go along with the other test code.

Also replace MockIPTablesManager by the same fake/mock implementation
already defined in pkg/datapath/fake/types.FakeIptablesManager.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

mhofstetter reviewed (approved):
Thanks!
